devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -250,12 +240,84 @@ public void execute() throws MojoFailureException, MojoExecutionException {
             //this stuff does not change
             // Do not include URIs in the manifest, because some JVMs do not like that
             StringBuilder classPathManifest = new StringBuilder();
-            StringBuilder classPath = new StringBuilder();
+            DevModeContext devModeContext = new DevModeContext();
+            for (Map.Entry<Object, Object> e : System.getProperties().entrySet()) {
+                devModeContext.getSystemProperties().put(e.getKey().toString(), (String) e.getValue());
+            }
 
             final AppModel appModel;
             try {
                 final LocalProject localProject = LocalProject
                         .resolveLocalProjectWithWorkspace(LocalProject.locateCurrentProjectDir(outputDirectory.toPath()));
+                //we need to establish a partial ordering of the projects (i.e. 'reactor build order')
+
+                List<AppArtifactKey> orderedProjects = new ArrayList<>();
+                HashSet<AppArtifactKey> toplace = new HashSet<>();
+                for (Map.Entry<AppArtifactKey, LocalProject> i : localProject.getWorkspace().getProjects().entrySet()) {
+                    toplace.add(i.getKey());
+                }
+                //TODO: there is probably a better algorithm than this to establish the partial ordering
+                //basically we just iterate and add anything to the list that has not dependencies in 'toplace'
+                //this has worst case performance of O(n^2), if there is a linear relationship between
+                //the modules and the original order is reversed. As N is generally fairly small we can live with this for now
```

Comment:
Theoretically, we might not need to include all the projects from the workspace. We could take the target project, navigate its dependencies and that should be it? I could look into that afterwards if you like.

Reply:
If the number of maven modules is big enough that the partial ordering algorithm becomes noticeable it would be a drop in the ocean. That TODO should be removed, I would be suprised if N is ever more than 100

Reply:
Yeah, it's not only the ordering though. They will be checked for changes, etc.

Reply:
ah, I see what you mean. I think as a first attempt it is ok to monitor the whole project and look at improving it later